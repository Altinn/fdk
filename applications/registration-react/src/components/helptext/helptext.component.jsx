import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Collapse } from 'reactstrap';
import _ from 'lodash';
import { withStateHandlers } from 'recompose';

import { getTranslateText } from '../../lib/translateText';
import localization from '../../lib/localization';
import './helptext.scss';

export const Helptext = props => {
  const {
    title,
    required,
    abstract,
    description,
    helptextItems,
    toggleShowAll,
    showAll
  } = props;

  const collapseClass = cx('fa', 'fdk-fa-left', {
    'fa-angle-double-down': !showAll,
    'fa-angle-double-up': showAll
  });

  const shortdesc = abstract || _.get(helptextItems, 'shortdesc');
  const descriptionText = description || _.get(helptextItems, 'description');

  return (
    <div className="fdk-reg-helptext mb-3 p-3">
      <div className="d-flex align-items-center">
        <h3>{title}</h3>
        {required && (
          <span className="fdk-badge badge badge-secondary ml-2">
            {localization.helptext.required}
          </span>
        )}
      </div>
      <div className="d-md-flex">
        {shortdesc && (
          <p
            className="m-0"
            dangerouslySetInnerHTML={{
              __html: getTranslateText(shortdesc).replace(
                new RegExp('\n', 'g'),
                '<br />'
              )
            }}
          />
        )}
        {descriptionText &&
          getTranslateText(descriptionText).trim() !== '' && (
            <button
              className="fdk-btn-no-border text-left p-0 ml-1 fdk-reg-helptext-more align-self-start"
              onClick={toggleShowAll}
            >
              <i className={collapseClass} />
              {showAll
                ? localization.helptext.less
                : localization.helptext.more}
            </button>
          )}
      </div>
      {descriptionText &&
        getTranslateText(descriptionText).trim() !== '' && (
          <Collapse className="mt-3" isOpen={showAll}>
            <p
              dangerouslySetInnerHTML={{
                __html: getTranslateText(descriptionText).replace(
                  new RegExp('\n', 'g'),
                  '<br />'
                )
              }}
            />
          </Collapse>
        )}
    </div>
  );
};

Helptext.defaultProps = {
  title: '',
  required: false,
  abstract: null,
  description: null,
  helptextItems: null,
  toggleShowAll: _.noop,
  showAll: false
};

Helptext.propTypes = {
  title: PropTypes.string,
  required: PropTypes.bool,
  abstract: PropTypes.string,
  description: PropTypes.string,
  helptextItems: PropTypes.object,
  toggleShowAll: PropTypes.func,
  showAll: PropTypes.bool
};

const enhance = withStateHandlers(
  ({ initialShowAll = false }) => ({
    showAll: initialShowAll
  }),
  {
    toggleShowAll: ({ showAll }) => e => {
      e.preventDefault();
      return { showAll: !showAll };
    }
  }
);

export default enhance(Helptext);
