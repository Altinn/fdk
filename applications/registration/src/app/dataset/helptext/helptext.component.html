<p class="fdk-input-helptext">
  <input type="checkbox" class="more-help-checkbox" id="{{name}}-more-help-button" />
  <span class="faerre-anbefalinger">
    {{resolvedShortDesc ? resolvedShortDesc : "Hjelpetekst er ikke klar for dette feltet"}}
  </span>
  <!-- To avoid whitespace between "Flere anbefalinger" and the . at the end,
  some tags below don't have whitespace between the closing tag and the next
  opening tag -->
  <label for="{{name}}-more-help-button" class="toggle-help-button" *ngIf=resolvedDescription>Flere anbefalinger</label>
  <span class="flere-anbefalinger">
    {{resolvedDescription}}
    <label for="{{name}}-more-help-button" class="toggle-help-button">FÃ¦rre anbefalinger</label>
  </span>.
</p>
