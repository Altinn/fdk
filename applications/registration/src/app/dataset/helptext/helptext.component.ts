import { Component, Input, Output, OnInit, EventEmitter } from '@angular/core';
import {HelptextsService} from "../helptexts.service";

@Component({
    selector: 'helptext',
    templateUrl: './helptext.component.html'
})

export class HelpText implements OnInit {
    @Input('text')
    public text: string;

    @Input('moretext')
    public moretext: string;

    @Input('name')
    public name: string;

    allHelptexts: any[];
    resolvedShortDesc: string = null;
    resolvedDescription: string = null;

  constructor(private helptextsService: HelptextsService) {
  }

  ngOnInit() {
    this.fetchHelpTexts().then(()=> {
      this.allHelptexts
        .filter(entry => entry.id == this.name)
        .forEach(entry => {
          this.resolvedShortDesc = entry.shortdesc;
          this.resolvedDescription = entry.description;
        })
    });
  }

  fetchHelpTexts() {
    return this.helptextsService.fetchHelptexts('nb').then(helptexts =>
      this.allHelptexts = helptexts);
  }
}
