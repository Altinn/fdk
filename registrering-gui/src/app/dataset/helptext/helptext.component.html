
<p class="input-helptext">
  {{text}}
  <input type="checkbox" class="more-help-checkbox" id="{{name}}-more-help-button" />
  <!-- To avoid whitespace between "Flere anbefalinger" and the . at the end,
  some tags below don't have whitespace between the closing tag and the next
  opening tag -->
  <label for="{{name}}-more-help-button" class="toggle-help-button">Flere anbefalinger</label><span class="flere-anbefalinger">
    {{moretext}}
    <label for="{{name}}-more-help-button" class="toggle-help-button">FÃ¦rre anbefalinger</label></span>.
</p>
