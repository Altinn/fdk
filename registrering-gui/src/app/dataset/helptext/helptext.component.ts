import { Component, Input, Output, OnInit, EventEmitter } from '@angular/core';
import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';

@Component({
    selector: 'helptext',
    templateUrl: './helptext.component.html'
})

export class HelpText implements OnInit {
    @Input('text')
    public text: string;

    @Input('moretext')
    public moretext: string;

    @Input('name')
    public name: string;


    constructor() {}

    ngOnInit() {
    }
}
