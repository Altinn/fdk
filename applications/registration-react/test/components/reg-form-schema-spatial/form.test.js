import React from 'react';
import { shallow } from 'enzyme';
import FormSpatial from '../../../src/components/reg-form-schema-spatial/form';
import helptext from '../../fixtures/helptext';

let defaultProps, wrapper;

beforeEach(() => {
  const { helptextItems } = helptext;
  defaultProps = {
    helptextItems: helptextItems,
  };
  wrapper = shallow(<FormSpatial {...defaultProps} />);
});


test('should render FormSpatial correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

