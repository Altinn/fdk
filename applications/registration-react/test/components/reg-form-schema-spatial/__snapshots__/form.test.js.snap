// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormSpatial correctly 1`] = `""`;
